modules/setting/service.go

```diff
@@ -103,6 +104,21 @@ func (a AllowedVisibility) ToVisibleTypeSlice() (result []structs.VisibleType) {
 	return result
 }
 
+// BuildEmailGlobs takes in an array of strings and
+// builds an array of compiled globs used to do
+// pattern matching in IsEmailDomainAllowed. A compiled list
+func BuildEmailGlobs(list []string) []glob.Glob {
+	var EmailList []glob.Glob
+
+	for _, s := range list {
+		if g, err := glob.Compile(s); err == nil {
+			EmailList = append(EmailList, g)
```

Comment:
I think errors should be logged, to indicate an incorrect glob in the configuration file.

Reply:
I've added what I think you're looking for, but tbh when I look at the current codebase it looks like gitea handles errors for glob compiles differently. Is there a preference here ?

Found code examples.


`log.Error("glob.Compile %s failed: %v", h, err)`
https://github.com/go-gitea/gitea/blob/main/modules/proxy/proxy.go#L72

vs 

`log.Info("Invalid glob expression '%s' (skipped): %v", expr, err)`
https://github.com/go-gitea/gitea/blob/main/modules/setting/indexer.go#L90

vs

`return nil, fmt.Errorf("'%s': Invalid glob pattern: %v", args[i], err)`
https://github.com/go-gitea/gitea/blob/main/cmd/embedded.go#L328

Reply:
> I've added what I think you're looking for, but tbh when I look at the current codebase it looks like gitea handles errors for glob compiles differently. Is there a preference here ?

Oh, that could be another PR to refactor those errors messages to a standard one. My preference goes to:

`log.Info("Invalid glob expression '%s' (skipped): %v", expr, err)` -> (with a small modification) `log.Warn("Invalid glob expression %q (skipped): %v", expr, err)`

As it also indicates the expression will be skipped.

Reply:
error could be returned

Reply:
Thanks, made the change.